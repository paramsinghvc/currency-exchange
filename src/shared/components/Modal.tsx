import React, { FC } from "react";
import ReactDOM from "react-dom";
import styled from "@emotion/styled";

import CrossIcon from "assets/Cross.svg";

const Backdrop = styled.section`
  position: absolute;
  top: 0;
  left: 0;
  width: inherit;
  max-width: inherit;
  height: inherit;
  background: rgba(255, 255, 255, 0.7);
  /* -webkit-filter: blur(12px);
  filter: blur(12px); */
`;

const Header = styled.header`
  display: flex;
  justify-content: space-between;
  padding: 20px;
  align-items: center;
`;

const CloseButton = styled.img``;

const Modal: FC<{
  open: boolean;
  headerText?: string;
  onClose: () => void;
}> = ({ children, open, headerText, onClose }) => {
  return open
    ? ReactDOM.createPortal(
        <Backdrop>
          <Header>
            <h3>{headerText}</h3>
            <CloseButton src={CrossIcon} role="button" onClick={onClose} />
          </Header>
          {children}
        </Backdrop>,
        document.body
      )
    : null;
};

export default Modal;
