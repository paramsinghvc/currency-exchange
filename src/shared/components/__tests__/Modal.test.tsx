import React from "react";
import Modal from "../Modal";
import renderer from "react-test-renderer";

describe("Should render Modal properly", () => {
  it("should return null open is false", () => {
    const modal = renderer.create(<Modal onClose={() => {}} headerText="EXCHANGE" open={false} />);
    expect(modal.toJSON()).toMatchSnapshot();
  });
});
