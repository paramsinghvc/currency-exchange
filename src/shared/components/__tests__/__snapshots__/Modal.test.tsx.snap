// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Modal properly should return null open is false 1`] = `null`;
